components/lib/crash/README.md

```diff
@@ -0,0 +1,155 @@
+# [Android Components](../../../README.md) > Libraries > Crash
+
+A generic crash reporter library that can report crashes to multiple services.
+
+Main features:
+
+* Support for multiple crash reporting services (included is support for [Sentry](https://sentry.io) and [Socorro](https://wiki.mozilla.org/Socorro)).
+* Support for crashes caused by uncaught exceptions.
+* Support for native code crashes (currently primarily focused on GeckoView crashes).
+* Can optionally prompt the user for confirmation before sending a crash report.
+* Support for showing in-app confirmation UI for non-fatal crashes.
+
+## Usage
+
+### Setting up the dependency
+
+Use Gradle to download the library from [maven.mozilla.org](https://maven.mozilla.org/) ([Setup repository]()):
+
+```Groovy
+implementation "org.mozilla.components:lib-crash:{latest-version}"
+```
+
+### Setting up crash reporting
+
+In the `onCreate()` method of your Application class create a `CrashReporter` instance and call `install()`:
+
+```Kotlin
+CrashReporter(
+    services = listOf(
+        // List the crash reporting services you want to use
+    )
+).install(this)
+```
+
+### Sending crash reports to Sentry
+
+⚠️ Note: The crash reporter library is compiled against the Sentry SDK but it doesn't require it as a dependency. The app using the component is responsible for adding the Sentry dependency to its build files in order to use Sentry crash reporting.
```

Comment:
Interesting, I was wondering whether this needs to be another concept and multiple implementation case. Here we're hiding the implementations from the class loader (not creating instances or referencing any static fields/methods). Don't see a problem other than being careful not to break this as we wouldn't detect it automatically.

Reply:
Yeah, that would be the most clean way. However it feels super over-engineered here. I'd have to create two separate gradle modules (+ concept module) and artifacts just for two files. Our first consumer will be Focus and Focus actually requires both (and other GV apps in the future probably too). Luckily we will still be able to move the code to new artifacts later if we decide so. :)